doc/users-guide/datapar/promotion.rst

```diff
@@ -0,0 +1,324 @@
+:title: Users Guide: Promotion
+
+.. _ug-promotion:
+
+Promotion: implicit data parallelism
+====================================
+
+In Chapel, *function promotion* (or simply *promotion*) refers to an
+implicit form of data parallelism that is triggered by passing a
+collection of values to an argument that expects a single value.  More
+precisely, when an array of type *t* (or a range or domain whose index
+type is *t*) is passed to a function argument of type *t*, that
```

Comment:
Is promotion of iterable-expressions (e.g. a leader/follower iterator) too low-level for this user's guide? I'd probably say "when an iterable expression yielding type *t* is passed to a function argument of type *t*, that function will be called in a data-paralle manner for all values yielded by the iterator."

One of the things that confused me the most when starting to learn Chapel is that a 'forall' expression (e.g. with the `[x in A]` syntax) actually creates an iterator rather than an array. While it's true that arrays are the common and easier-to-understand case here, it seems that being clear about how it works with iterators would be nice.